### PR TITLE
misc: replace all "Metadatas" with "Metadata"

### PR DESCRIPTION
**Pull request by swalahamani**
## Breaking changes

- We replaced all API containing `metadatas` with `metadata`. This includes but is not limited to:
  - `themeConfig.metadatas`
  - `Layout.Props.searchMetadatas`
  - `@theme/SearchMetadatas`
  
  If you use any of these APIs, you need to update accordingly.
  **Why?** Because `metadatas` is grammatically incorrect. We are not going to pretend like we are grammar masters who made this API with a surprising purpose—we are just wrong, and now we fix it.

## Motivation

The plural form of **Metadata** is **Metadata** itself and not **Metadatas**.
Please refer: [Wordhippo](https://www.wordhippo.com/what-is/the-plural-of/metadata.html#:~:text=Answer,of%20metadata%20is%20also%20metadata.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

N/A as this is just a spelling correction.

## Related PRs
 
N/A

### CODE REVIEW

**Comment by slorber** [comment]
It's my fault, used to think metadatas was correct 😅 

I'm ok to change it everywhere, including a breaking change in theme API considering we'd rather do that before 2.0

**Comment by Josh-Cena** [comment]
Actually, I did a random chat with someone else and they said `metadatas` is acceptable :P

> If you have a concept of `metadata`, and `metadatas` is an array of that, I think it's perfectly fine
> I could see an argument that `metadata` is a group of `metadatums`, and `metadatas` is a group of multiple `metadatums`

**Comment by slorber** [comment]
🤪

File: packages/docusaurus-theme-classic/src/validateThemeConfig.ts [hunk 1]
```diff
@@ -264,9 +264,9 @@ const ThemeConfigSchema = Joi.object({
   colorMode: ColorModeSchema,
   image: Joi.string(),
   docs: DocsSchema,
-  metadatas: Joi.array()
+  metadata: Joi.array()
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/validateThemeConfig.ts:267]: Perhaps we should add such a message for users who have used `metadatas`?

```suggestion
  metadatas: Joi.any().forbidden().messages({
    'any.unknown':
      'themeConfig.metadatas has been renamed as themeConfig.metadata',
  }),
```

Comment (Josh-Cena) [reply packages/docusaurus-theme-classic/src/validateThemeConfig.ts:267]: Yes, I will do that. Thanks for reminding

**Review by slorber — APPROVED** [review]
LGTM thanks 👍

**Pull request merged**